src/Algebra/Construct/Centre/Group.agda

```diff
@@ -0,0 +1,85 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Definition of the centre of a Group
+------------------------------------------------------------------------
+
+{-# OPTIONS --safe --cubical-compatible #-}
+
+open import Algebra.Bundles
+  using (Group; AbelianGroup; RawMonoid; RawGroup)
+
+module Algebra.Construct.Centre.Group {c ℓ} (group : Group c ℓ) where
+
+open import Algebra.Core using (Op₁)
+open import Algebra.Morphism.Structures
+open import Algebra.Morphism.GroupMonomorphism using (isGroup)
+open import Function.Base using (id; const; _$_)
+
+
+private
+  module G = Group group
+
+open import Relation.Binary.Reasoning.Setoid G.setoid as ≈-Reasoning
+open import Algebra.Properties.Group group using (∙-cancelʳ)
+
+
+------------------------------------------------------------------------
+-- Definition
+
+-- Re-export the underlying sub-Monoid
+
+open import Algebra.Construct.Centre.Monoid G.monoid as Z public
+  using (Center; ι; ∙-comm)
+
+-- Now, can define a commutative sub-Group
+
+_⁻¹ : Op₁ Center
+g ⁻¹ = record
+  { ι = ι g G.⁻¹
+  ; central = λ k → ∙-cancelʳ (ι g) _ _ $ begin
+      (ι g G.⁻¹ G.∙ k) G.∙ (ι g) ≈⟨ G.assoc _ _ _ ⟩
```

Comment:
these assoc/cong, cong/assoc lines are exactly what the push/pull combinators from `agda-categories` were all about. They help take this proof from 7 lines down to 4. This comment is more of an educational comment than a "please change this".

Reply:
Of course! I hadn't previously appreciated 'the point' of these combinators (other than in the abstract), and hadn't studied their use in anger, cf. my interpolation of `Quasigroup`/`Loop` properties into those for `Group`. Sheer intellectual blinkers on my part in writing the 4 lines 'from first principles' ... so do please feel free to offer the 4 line proof as a GitHub `suggestion`!

Elsewhere, I have echoing on my mind an observation made somewhere else in GitHub (how to find such things!?) that the `Group` case can be further streamlined by observing something like (is this true? scratches head for a moment): any `MonoidHomomorphism` between (the underlying `Monoid`) `Group`s must, in fact, be a `GroupHomomorphism`, also by a 'cancellability' argument...?

Reply:
As for 'educational': thanks for the nudge! Latest commit simplifies...

Reply:
But in the spirit of the above comment (lifted out as issue #2889 now), and knowing that `ι` defines a monoid *mono*morphism, can we further simplify the construction of `Centre.Group`? Or is that the 'real' semantic content of the appeal to `GroupMonomorphism.isGroup` as the 'smart constructor' I'm looking for?